src/commands/lightning/preview/app.ts

```diff
@@ -202,46 +204,68 @@ export default class LightningPreviewApp extends SfCommand<void> {
+    // todo: For the desktop scenario, should we print a message to the terminal screen to inform the user
+    // that they should configure their browser to allow-list this connection to localhost, or does it suffice
+    // to just mention that in the documentation?
+    // if (ldpServerUrl.toLowerCase().startsWith('wss://')) {
+    //   print message to screen
+    // }
 
     await this.config.runCommand('org:open', launchArguments);
   }
 
   private async mobilePreview(
     platform: Platform.ios | Platform.android,
-    ldpServerUrl: string,
-    appName?: string,
-    appId?: string,
-    deviceId?: string,
-    logger?: Logger
+    sfdxProjectRootPath: string,
+    appName: string | undefined,
+    appId: string | undefined,
+    deviceId: string | undefined,
+    logger: Logger
   ): Promise<void> {
     try {
-      // 1. Verify that user environment is set up for mobile (i.e. has right tooling)
+      // Verify that user environment is set up for mobile (i.e. has right tooling)
       await this.verifyMobileRequirements(platform, logger);
 
-      // 2. Fetch the target device
+      // Fetch the target device
       const device = await PreviewUtils.getMobileDevice(platform, deviceId, logger);
       if (!device) {
         throw new Error(messages.getMessage('error.device.notfound', [deviceId ?? '']));
       }
 
-      // 3. Boot the device if not already booted
+      // Boot the device if not already booted
       this.spinner.start(messages.getMessage('spinner.device.boot', [device.toString()]));
       const resolvedDeviceId = platform === Platform.ios ? (device as IOSSimulatorDevice).udid : device.name;
       const emulatorPort = await PreviewUtils.bootMobileDevice(platform, resolvedDeviceId, logger);
       this.spinner.stop();
 
-      // 4. Generate self-signed certificate and wait for user to install it
-      // TODO: update the save location to be the same as server config file path
-      this.spinner.start(messages.getMessage('spinner.cert.gen'));
-      const certFilePath = PreviewUtils.generateSelfSignedCert(platform, '~/Desktop/cert');
-      this.spinner.stop();
-      await LightningPreviewApp.waitForUserToInstallCert(platform, device, certFilePath);
+      // Configure certificates for dev server secure connection
+      let secureConnectionFiles = await ConfigUtils.getSecureConnectionFiles();
+      if (!secureConnectionFiles?.pemCertFilePath || !secureConnectionFiles.pemKeyFilePath) {
+        this.spinner.start(messages.getMessage('spinner.cert.gen'));
+        // Generate self-signed certificate
+        // todo: should the cert files be saved at the root of sfdx project or inside .sf folder of the project?
+        secureConnectionFiles = PreviewUtils.generateSelfSignedCert(platform, sfdxProjectRootPath);
+        // Configure the local dev server with certFilePath & keyFilePath
+        await ConfigUtils.writeSecureConnectionFiles(secureConnectionFiles);
```

Comment:
@ravijayaramappa How did we decide we were going to handle the interoperability of different user agents needing HTTP vs. HTTPS? That would be a restart of the dev server at this point, yes? I think we need to offset this "write" with a "delete" when HTTPS is _not_ desired. Because the configuration is persisted.

Thinking through that lens, @maliroteh-sf rather than inferring the HTTPS configuration (or lack thereof) based on anticipated use case, we should probably create some helper functionality the decides whether or not we need cert configuration, based on the assessment of (or similar logic to) the WS URI we need.

Reply:
Hmmm.... I'm thinking that we need to generate the cert files only once. The cert is valid for ~2 years and can be reused over and over for different cases. For example, if the user wants to preview on iOS and then on Android, they should be able to use the same cert file.... and once they configure their ios/android device they won't need to do that again for another 2 years since we are reusing the same cert file.

On the same note, I think that the cert file should `not` be stored at the root of an SFDX project. This is b/c by doing so we are essentially generating new cert file for different SFDX projects.

 So suppose the user has already generated a cert file for SFDX project1 and installed it on their iOS device. Then they want to preview another SFDX project but on the same iOS device. If we generate a new cert file for SFDX project2, now the user needs to uninstall the previous cert file from the device and install the new one before proceeding.
 
 If we generate the cert files at a global location (say `~/.sf`) then we can reuse the same cert for the next 2 years. We can then follow the logic below:
 
 1. If there are no certificates in `~/.sf` then go ahead and generate them (`localhost.der` + `localhost.pem` + `localhost-key.pem`) regardless of whether we are running for desktop or mobile. These cert files can be used for the next 820 days
 
 2. Generate the LDP server url.
 
 3. If LDP server url starts with `wss://` then we need a secure connection, so use the generated certs in `~/.sf`
 
 4. Refactor the code [here](https://github.com/salesforcecli/plugin-lightning-dev/blob/main/src/lwc-dev-server/index.ts#L82-L102) to something like below, and this way we don't even need to use `configMeta + configUtils` to save the location of the cert files to the config of an SFDX project and it won't be persisting, b/c instead we will create an instance of `ServerConfig` on the fly each time we want to start the dev server:
 ```
 export async function startLWCServer(rootDirOrConfig: string | ServerConfig, logger: Logger): Promise<LWCServer> {
  const finalConfig = 
    (typeof(rootDirOrConfig) === 'string') 
      ? await createLWCServerConfig(rootDirOrConfig, logger)
      : rootDirOrConfig;
  
  logger.trace(`Starting LWC Dev Server with config: ${JSON.stringify(finalConfig)}`);
  let lwcDevServer: LWCServer | null = await startLwcDevServer(finalConfig);
  .
  .
  .
}

Reply:
That's a good point, with regards to having to reconfigure certificate trust all the time for devices on the host machine. For that reason we probably do want to save these values globally.

I'm thinking out loud about a hybrid model of what you suggest, where we both:

- Generate / persist the certificate files once, in a global location.
- _Copy_ these to each _local_ project as necessary, from the global configuration.

The reason I'm thinking of the hybrid potential is that the user still needs to "drag and drop" these certificates to their device(s). In that respect, a) I typically don't like to send users down a "hidden" (`.<dir>`) directory path to search for files, and b) having the files _specifically_ close to their project will make it easier for them to find, because of the association with the project they're working on.

In that scenario we could even persist the (base64-encoded, as necessary) contents of the certificate files themselves within our existing configuration model, just with the `--global` scope for that data, and rehydrate local certificate files from there. Then we don't have to mess around with direct file operations outside of our project directory scope. Anyway, it's a thought I had. I don't know if it's a good one. 🙂

But overall, I think your reasoning on this point is good, and it seems like a sound approach to me.